File: 0283_Move Zeroes/MoveZeroes2.cs
public class Solution {
   public void MoveZeroes(int[] nums) {
       if(nums == null || nums.Length <= 1) return;
       var start = 0;
       var end = 0;
       while(end < nums.Length)
       {
           if(nums[end] == 0) end++;
           else{
               nums[start++] = nums[end++];
           }
       }

       for(int i=start;i<nums.Length;i++)
           nums[i] = 0;
   }
}

File: 0283_Move Zeroes/MoveZeroes_2.java
class Solution {
    public void moveZeroes(int[] nums) {
        int p = 0;
        for(int i=0;i<nums.length;i++) {
            if(nums[i] != 0) {
                nums[p++] = nums[i];
            }
        }
        
        while(p < nums.length)
            nums[p++] = 0;
    }
}